mempool/src/shared_mempool/types.rs

```diff
@@ -188,8 +188,27 @@ impl fmt::Display for CommitNotification {
 
 #[derive(Debug)]
 pub struct CommitResponse {
-    /// error msg if applicable - empty string if commit was processed successfully by mempool
-    pub msg: String,
+    pub error: bool,
+    /// The error message if `error` is true.
+    pub error_message: Option<String>,
```

Comment:
`error` field seems redundant, it can be derived from the Option?

Reply:
Sure, although just having an optional error message doesn't seem as clear as having an error field too. For example, what if there is no error message but an error encountered? Personally, I prefer the error bool and message. But I don't feel super strongly, here. Feel free to override and I'll change it 😄

Reply:
I would recommend pub success: bool it is the more common approach

also note that this is a breaking change, is it not? What's your upgrade and rollout plan :P

Reply:
I've updated this to use the "success: bool" terminology, and have an optional error message iff success is false.

Regarding this being a breaking change, I'm not sure that I follow. From what I've seen CommitResponse is only used in intra-node communication (i.e., between components within a node). These are the use scenarios I can see:
- Consensus commits new txs and tells state sync using StateSyncClient. This sends a CommitNotification to StateSyncCoordinator. StateSyncCoordinator then handles the commit and sends a CommitResponse back to StateSyncClient.
- A node sends a chunk response to this node. `process_commit` is called (iff the chunk is valid). A CommitResponse is created, but not sent anywhere.
- On any `process_commit` in StateSyncCoordinator, mempool is notified by a CommitNotification. When mempool gets a CommitNotification it handles it in `process_state_sync_request` and creates a CommitResponse to send back to StateSyncCoordinator (via a callback). StateSyncCoordinator just checks that the CommitResponse is successful.

Outside of these usages, I don't see any flows that would make this breaking between nodes. Unless you were thinking we may update components (e.g., consensus, mempool, state sync) independently on releases? (Which sounds hard 😄)